sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -236,8 +236,8 @@ abstract class BaseRelation {
 
   /**
    * Given an array of [[Filter]]s, returns an array of [[Filter]]s that this data source relation
-   * cannot handle.  Spark SQL will apply all returned [[Filter]]s against rows returned by this
-   * data source relation.
+   * may not be able to apply to every row. Spark SQL will apply all returned [[Filter]]s against
+   * rows returned by this data source relation.
```

Comment:
Returns the list of filters that this datasource may not be able to handle. These returned filters will be evaluated by Spark SQL after data is output by a scan.  By default, this function will return all filters, as it is always safe to double evaluate a filter.  However, specific implementations can override this function to avoid double filtering when they are capable of processing a filter internally.